lightning/src/ln/channelmanager.rs

```diff
@@ -951,11 +951,121 @@ where
 	fn get_cm(&self) -> &ChannelManager<M, T, ES, NS, SP, F, R, L> { self }
 }
 
-/// Manager which keeps track of a number of channels and sends messages to the appropriate
-/// channel, also tracking HTLC preimages and forwarding onion packets appropriately.
+/// A lightning node's channel state machine and payment management logic.
 ///
-/// Implements [`ChannelMessageHandler`], handling the multi-channel parts and passing things through
-/// to individual Channels.
+/// [`ChannelManager`] is parameterized by a number of components, which in aggregate facilitate
```

Comment:
Yeah, this goes back to how we want to breakdown the organization of these docs across modules and structs as mentioned in the PR description. The draft has everything on `ChannelManager`, but my feeling is we should minimize those docs and move the sectioned breakdown to higher-level docs, likely on a module akin to how `Iterator` is documented in the `std` library.

So I can see moving everything after this list to the module-level, linking to it from here. I'm fine with also putting a short mention of `ChannelMonitor` here and linking to the particular module-level section doc that goes into detail.

Reply:
I do think the number of foundational components needed to get a `ChannelManager` up a running is worth noting. I've found it does help devs with their mental models and how things connect together

Reply:
Yip. Agree with @ConorOkus on knowing what it makes use of upfront. At least in my experience it helps to know what it _is_ (including what it's composed of) first and then more deeply go into what it _does_ and _how_, then all the other caveats to take note of.

Reply:
@dunxen Any opinion on whether most of the docs should move to the module-level as seems to be common in Rust standard library docs?

Reply:
Might expand the module docs quite a bit. Otherwise either seems fine if that’s not a concern (or if it’s bare now anyway)

Reply:
For me as a dev, It would be helpful to expand the first sentence. 
Firstly, why is it 2 things: state machine and payment management? I always refer to it to just state machine, but then maybe I am wrong here.
Secondly, as a state machine it would be helpful to know the possible states and actions that change the state, where is the state being held, saved, loaded?
Lastly, what logic does this state machine implement? What logic goes into the parameters and what is in the state machine and why.

Reply:
Thanks for the feedback! I'm not sure exactly how to address these yet, but wanted to respond in case it helps you or others see a way to improve these docs. We'll likely land this as-is but I'm definitely open to improvements in follow-ups. Contributions are more than welcome, too.

> For me as a dev, It would be helpful to expand the first sentence. Firstly, why is it 2 things: state machine and payment management? I always refer to it to just state machine, but then maybe I am wrong here.

While there may be some refactoring that would allow us to have a cleaner separation of channel state machine from payment management, they are intrinsically linked as payment-related messages cause channel state transitions. Some of which may result in payment completion, failure, retry, or timeout.

> Secondly, as a state machine it would be helpful to know the possible states and actions that change the state, where is the state being held, saved, loaded?

That seems more like internal developer docs. The goal of `ChannelManager` is to abstract away the details of the state machine from developers. That said, there are probably higher-level states exposed in `ChannelDetails` and `RecentPaymentDetails` that could be useful to better document there. I believe we are at least linking to those docs here albeit indirectly through methods that return those types.

> Lastly, what logic does this state machine implement? What logic goes into the parameters and what is in the state machine and why.

The Lightning BOLT specification. :) Or at least some of it. Kidding aside, I think we want to err on the side of brevity here, but documentation on the linked parameterized types are in many instances ripe for going into further details, if they haven't already. I'll be the first to admit our docs need some work. Hopefully this is just a first step.

Reply:
> > Secondly, as a state machine it would be helpful to know the possible states and actions that change the state, where is the state being held, saved, loaded?
> 
> That seems more like internal developer docs. The goal of `ChannelManager` is to abstract away the details of the state machine from developers. That said, there are probably higher-level states exposed in `ChannelDetails` and `RecentPaymentDetails` that could be useful to better document there. I believe we are at least linking to those docs here albeit indirectly through methods that return those types.
> 

A further point here is how such transitions result in `Event` generation. We do go into that in the examples, but yes those could possibly use some higher-level docs as it relates to state transitions.

Reply:
> That said, I do agree the first sentence could be a bit more user-relevant, and maybe get a second sentence in the first paragraph. Maybe the "Channelmanager...facilitate sending, forwarding, and receiving payments through lightning channels" should move to the first sentence, and the second paragraph intro can just be "ChannelManager is parameterized"?

Ah, ok, that sounds good. See latest push.